src/main/java/tools/map/xml/creator/MapXmlHelper.java

```diff
@@ -456,438 +461,328 @@ static void clearGameSettings() {
-      final Node productionRule = productionChildNodes.item(i);
-      if (productionRule.getNodeName().equals(XML_NODE_NAME_PRODUCTION_RULE)) {
-        parseProductionRuleNode(productionRule.getChildNodes());
-      } else if (productionRule.getNodeName().equals(XML_NODE_NAME_PRODUCTION_FRONTIER)) {
-        final String playerName =
-            productionRule.getAttributes().getNamedItem(XML_ATTR_ATTACHMENT_NAME_NAME).getNodeValue().substring(10);
-        final ArrayList<String> frontierRules = new ArrayList<>();
-        final NodeList productionFrontierChildNodes = productionRule.getChildNodes();
-        for (int j = 0; j < productionFrontierChildNodes.getLength(); ++j) {
-          final Node productionFrontierChildNode = productionFrontierChildNodes.item(j);
-          if (productionFrontierChildNode.getNodeName().equals(XML_NODE_NAME_FRONTIER_RULES)) {
-            frontierRules
-                .add(productionFrontierChildNode.getAttributes().getNamedItem(XML_ATTR_ATTACHMENT_NAME_NAME)
-                    .getNodeValue().substring(3));
-          }
-        }
-        getProductionFrontiersMap().put(playerName, frontierRules);
+      @Override
+      protected void handlePlayer(Attributes attributes) {
+        final String playerNameAttr = attributes.getValue(XML_ATTR_PLAYER_NAME_NAME);
+        getPlayerNames().add(playerNameAttr);
+        getPlayerInitResourcesMap().put(playerNameAttr, 0);
+        // TODO: add logic for optional value
+        // attrMapPlayer.get("optional")
+      }
+
+      @Override
+      protected void handleAlliance(Attributes attributes) {
+        getPlayerAllianceMap().put(attributes.getValue(XML_ATTR_ALLIANCE_NAME_PLAYER),
+            attributes.getValue(XML_ATTR_ALLIANCE_NAME_ALLIANCE));
+      }
+
+      @Override
+      protected void handleUnit(Attributes attributes) {}
+
+      @Override
+      protected void handleDelegate(Attributes attributes) {
+        getGamePlaySequenceMap().put(attributes.getValue(XML_ATTR_ATTACHMENT_NAME_NAME),
+            Arrays.asList(
+                attributes.getValue(XML_ATTR_ATTACHMENT_NAME_JAVA_CLASS).replace(TRIPLEA_JAVA_CLASS_DELEGATE_PATH, ""),
+                attributes.getValue(XML_ATTR_STEP_NAME_DISPLAY)));
+      }
+
+      @Override
+      protected void handleStep(Attributes attributes) {
+        final String maxRunCount = attributes.getValue(XML_ATTR_STEP_NAME_MAX_RUN_COUNT);
+        getPlayerSequenceMap().put(attributes.getValue(XML_ATTR_STEP_NAME_NAME),
+            Triple.of(attributes.getValue(XML_ATTR_STEP_NAME_DELEGATE),
+                Strings.nullToEmpty(attributes.getValue(XML_ATTR_STEP_NAME_PLAYER)),
+                (maxRunCount == null ? 0 : Integer.parseInt(maxRunCount))));
+      }
+
+      @Override
+      protected void handlePlayerProduction(Attributes attributes) {}
+
+      @Override
+      protected void handleCost(Attributes attributes) {
+        costQuantity = attributes.getValue(XML_ATTR_RESULT_NAME_QUANTITY);
```

Comment:
Looks like our handlers are mostly doing just this:
` costQuantity = attributes.getValue(XML_ATTR_RESULT_NAME_QUANTITY);`
That makes it seem like we can really simplify with the open close handler list. going that route as previously described, we might be able to move all xml name constants to the individual handlers, and would no longer need the `public static final String` values anymore at all.